src/module/rolls/roll.ts

```diff
@@ -0,0 +1,57 @@
+// TODO:
+// - Convert to HTML for chat messages
+// - Rehydrate from chat message (will be needed for rerolls)
+// - Options input covers all cases
+// - Rolling that plays nicer with DF Manual Rolls (all in one go, not {d6+N,d10,d10})
+// - Rerolls update chat message
+
+export enum HIT_TYPE {
```

Comment:
terminology nitpick: should it be "hit type" if one of the types isn't a hit?

there isn't a strict terminology ethe most common term for these in IS/SF is "outcome", and this is what dataforged uses; in this context my suggestion would be `MOVE_OUTCOME`.

tangent (not sure it'd be useful here, i just think it's neat): in a couple of implementations ive worked on, i've played with enums like the following:

```typescript
export enum MOVE_OUTCOME {
  Miss = 0,
  WeakHit = 1,
  StrongHit = 2
}
```

what's nice about this is that the numeric value is actually *the number of challenge dice beaten by the action score/progress score*. can't remember if this is an idea i hit on or if i stole it from something else, and ultimately doesn't save *that* much code, but it might feel a little more elegant. ;)

Reply:
Settled on `ROLL_OUTCOME` for a name, because this isn't just about moves. I'll consider the numeric values.

Reply:
> Settled on `ROLL_OUTCOME` for a name, because this isn't just about moves. I'll consider the numeric values.

i might just be spacing on something obvious, but... what mechanical entities other than a move that can have a miss/weak hit/strong hit outcome in ironsworn? even if a "generic"/"unassigned" action roll/progress roll is being made as far as the module is concerned, the roll is still for a move, isn't it?

on the other hand, non-move rolls do exist - oracles are the most prominent example.

Reply:
The Starforged book does red highlighting on the terms "action roll" (p32) and "progress roll" (p43). What words should we use to describe the outcome of two kinds of rolls?

And let's not discount players going off-book. There's nothing saying you can _only_ roll because of moves. If you want to just roll +wits to see if your PC knows a fact, apropos of no move whatsoever, the book doesn't tell you to never do that. So this thing won't be used only with moves.